Cubical/HITs/Pushout/KrausVonRaumer.agda

```diff
@@ -0,0 +1,147 @@
+{-
+
+An induction principle for paths in a pushout, described in
+Kraus and von Raumer, "Path Spaces of Higher Inductive Types in Homotopy Type Theory"
+https://arxiv.org/abs/1901.06022
+
+-}
+
+{-# OPTIONS --cubical --safe #-}
+module Cubical.HITs.Pushout.KrausVonRaumer where
+
+open import Cubical.Foundations.Everything
+open import Cubical.Data.Sigma
+open import Cubical.HITs.Pushout.Base
+
+private
+  flippedHomotopyIsInverse : ∀ {ℓ} {A : Type ℓ} {f : A → A} (p : ∀ a → f a ≡ a)
+    (a : A) → Path (f a ≡ f a) (λ i → p (p a (~ i)) i) refl
+  flippedHomotopyIsInverse {f = f} p a j i =
+    hcomp
+      (λ k → λ {
+        (i = i0) → f a;
+        (i = i1) → p a (j ∧ ~ k);
+        (j = i0) → p (p a (~ i)) i;
+        (j = i1) → p a (i ∧ ~ k)})
+      (p (p a (~ i ∨ j)) i)
+
+  interpolate : ∀ {ℓ} {A : Type ℓ} {x y z : A} (q : y ≡ z)
+    → PathP (λ i → x ≡ q i → x ≡ z) (_∙ q) (idfun _)
+  interpolate q i p j =
+    hcomp
+      (λ k → λ
+        { (j = i0) → p i0
+        ; (j = i1) → q (i ∨ k)
+        ; (i = i1) → p j
+        })
+      (p j)
+
+  interpolateCompPath : ∀ {ℓ} {A : Type ℓ} {x y : A} (p : x ≡ y) {z : A} (q : y ≡ z)
+   → (λ i → interpolate q i (λ j → compPath-filler p q i j)) ≡ refl
+  interpolateCompPath p =
+    J (λ z q → (λ i → interpolate q i (λ j → compPath-filler p q i j)) ≡ refl)
+      (flippedHomotopyIsInverse (λ p i j → compPath-filler p refl (~ i) j) p)
+
+module ElimL {ℓ ℓ' ℓ'' ℓ'''} {A : Type ℓ} {B : Type ℓ'} {C : Type ℓ''}
+  {f : A → B} {g : A → C} {b₀ : B}
+  (P : ∀ b → Path (Pushout f g) (inl b₀) (inl b) → Type ℓ''')
+  (Q : ∀ c → Path (Pushout f g) (inl b₀) (inr c) → Type ℓ''')
+  (r : P b₀ refl)
+  (e : (a : A) (q : inl b₀ ≡ inl (f a)) → P (f a) q ≃ Q (g a) (q ∙ push a))
+  where
+
+  Codes : (d : Pushout f g) (q : inl b₀ ≡ d) → Type ℓ'''
+  Codes (inl b) q = P b q
+  Codes (inr c) q = Q c q
+  Codes (push a i) q =
+    Glue
```

Comment:
Can't you use `ua` here?

Reply:
Not as it is, the line being extended is not constant.